lib/mongo/mongo-plugin.js

```diff
@@ -71,16 +71,34 @@ function mongoRegister(fastify, schemaMaps) {
     },
   } = fastify
   const factoryChecker = checkFactory(fastify)
+  const autoEncryption = generateEncryptionConfig(fastify, schemaMaps)
+
   return (type) => {
     const pkFactory = pkFactories[type]
+
     factoryChecker(pkFactory, type)
     fastify.register(fastifyMongodb, {
       url: MONGODB_URL,
       pkFactory,
-      autoEncryption: generateEncryptionConfig(fastify, schemaMaps),
+      autoEncryption,
       name: getDatabaseNameByType(type),
       maxIdleTimeMs: MONGODB_MAX_IDLE_TIME_MS,
     })
+
+    // in case the encryption is enabled, then another Mongo client without auto-encryption
+    // is necessary to create Mongo Views (only auto-decryption is enabled)
+    if (autoEncryption) {
```

Comment:
should this client be created only if `type === view` ?

moreover, I'd add a dedicated test to this double client feature if possible

Reply:
Unfortunately, I don't have a direct access to the type of the collection. In fact, the `type` that arrives here represents whether the internal MongoDB `_id` is an ObjectId or a String.

Although creating the client only when at least a view definition is configured would be a better solution, without a large refactor of the code base, I believe the current solution should fit the bill.

Regarding the tests, this part is already covered by the existing tests with the encryption enabled, since without this newer portion of code they would fail with the following error:

    TypeError: csfle "analyze_query" failed: Creating a view is not supported with automatic encryption [Error 2, code 7501300]

Reply:
Thanks for the explanation, I was tricked by the variable `type` which I thought represented the collection's type. As a small change, I would change the variable of that callback to a more specific `idType`